OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -124,6 +126,12 @@ public void Killed(Actor self, AttackInfo e)
 				attackerStats.KillsCost += cost;
 				defenderStats.DeathsCost += cost;
 			}
+
+			if (self.Info.HasTraitInfo<GivesExperienceInfo>() && self.Info.TraitInfo<GivesExperienceInfo>().GivesPlayerExperience)
```

Comment:
Please cache the trait lookup with `self.Info.TraitInfo<GivesExperienceInfo>()` and a `!= null` check to makes this less wasteful regarding performance.